src/views/Products.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useEffect, useCallback } from 'react'
+import ProductsSidebar from '../components/ProductsSidebar'
+import ProductsJson from '../mocks/en-us/products.json'
+import '../styles/Products.scss'
+import ProductItem from '../components/FeaturedProducts/ProductItem'
+import ProductsPagination from '../components/ProductsPagination'
+import Spinner from '../components/Spinner'
+import { Link, useNavigate, useSearchParams } from 'react-router-dom'
+import { useProducts } from '../utils/hooks/useProducts'
+import { useSelector } from 'react-redux'
+
+export default function Products () {
```

Comment:
the logic in this component could be refined, I think it is doing to much, some of the logic could be moved to the children or small components that only do one thing